src/atomate2/abinit/sets/core.py

```diff
@@ -0,0 +1,237 @@
+"""Module defining core Abinit input set generators."""
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import Callable
+
+import numpy as np
+from abipy.abio.factories import (
+    dos_from_gsinput,
+    ebands_from_gsinput,
+    ion_ioncell_relax_input,
+    nscf_from_gsinput,
+    scf_input,
+)
+from abipy.abio.input_tags import MOLECULAR_DYNAMICS, NSCF, RELAX, SCF
+from abipy.abio.inputs import AbinitInput
+from pymatgen.core import Structure
+from pymatgen.io.abinit import PseudoTable
+from pymatgen.io.abinit.abiobjects import KSampling
+
+from atomate2.abinit.sets.base import AbinitInputGenerator
+
+__all__ = [
+    "StaticSetGenerator",
+    "NonSCFSetGenerator",
+    "RelaxSetGenerator",
+]
+
+
+GS_RESTART_FROM_DEPS: tuple = (f"{SCF}|{RELAX}|{MOLECULAR_DYNAMICS}:WFK|DEN",)
+
+
+@dataclass
+class StaticSetGenerator(AbinitInputGenerator):
+    """Common class for ground-state generators."""
+
+    calc_type: str = "static"
+    factory: Callable = scf_input
+    restart_from_deps: tuple = field(
+        default_factory=lambda: tuple(GS_RESTART_FROM_DEPS)
```

Comment:
No need for the default_factory stuff here and in other generators where you define `restart_from_deps`. That's because tuples are immutable anyway, so you can just definitely this as:
```python
restart_from_deps: tuple = (GS_RESTART_FROM_DEPS, )
```